### ISSUE TITLE
Add Azure Kubernetes Service AKS

### ISSUE THREAD

**Issue by NissesSenap** [infracost/infracost#591]
Pricing page: https://azure.microsoft.com/en-gb/pricing/details/kubernetes-service/

Make it possible to calculate how much a AKS cluster costs.

A basic graphql query to get info about AKS pricing.
```
query {
  products(
    filter: {
      vendorName: "azure",
      service: "Azure Kubernetes Service",
      region: "centralus",
      attributeFilters: []
    },
  ) {
    attributes { key, value }
    prices { USD }
  }
}
```

**Comment by alikhajeh1** [comment]
Thanks for adding this! I haven't used [`azurerm_kubernetes_cluster`](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster) or [`azurerm_kubernetes_cluster_node_pool`](azurerm_kubernetes_cluster_node_pool) before but it look similar to how we added the Google Kubernetes Engine service in that the cluster has a node pool, see `internal/providers/terraform/google/container_cluster.go` and `internal/providers/terraform/google/container_node_pool.go` (and their corresponding [usage-file](https://github.com/infracost/infracost/blob/master/infracost-usage-example.yml) params). For that, we needed re-used the cost components from the compute and disk resources. The [Azure price calculator](https://azure.microsoft.com/en-gb/pricing/calculator/?service=kubernetes-service) shows a similar thing (nodes and managed disks, both of which we support already but we'd need to refactor/re-use). We can start with the pay-as-you-go prices.

It looks like the cluster itself just has one cost component:
- If `sku_tier=Paid`, add the cost component name "Uptime SLA" with units "cluster-hours".

If this is the first time you're adding a resource to Infracost, you might prefer to add a simpler resource such as [this](https://github.com/infracost/infracost/issues/596) to go through the end-to-end process as an onboarding step.

**Comment by aliscott** [comment]
Here's the pricing I've found for AKS.

We might want to add support for the following services before we tackle this:
* [Azure Load Balancers](https://azure.microsoft.com/en-us/pricing/details/load-balancer/)
* [Azure DNS](https://azure.microsoft.com/en-gb/pricing/details/dns/#pricing)
* [Azure Monitoring](https://azure.microsoft.com/en-gb/pricing/details/monitor/)

If not, then I've split this into initial support and future support, so we can add a baseline level of support for this initially.

### Initial support

**azurerm_kubernetes_cluster**

Cost components:
* `Uptime SLA`
  * Unit: `hours`.
  * Applicable if `sku_tier == Paid`

Sub resources:
* `default_node_pool` - For this we will want to re-use some of the methods from the virtual machine resources
  * This has the same cost components and subresources as the virtual_ achine resources, e.g: `Instance usage (A2 v2, pay as you go)`
  * The  `os_disk` subresource applies if `os_disk_type == Managed`
  * We should multiply the quantities here by `node_count` (See the [MultiplyQuantities function](https://github.com/infracost/infracost/blob/3215fe71fd4c1d434596b58977a2c51bfe8a5fc6/internal/schema/resource.go#L100)). We should also have a usage key: `default_node_pool.nodes` which overrides this.

**azurerm_kubernetes_cluster_node_pool**

This has the same cost components / subresources as `default_node_pool`.

### Future support

* Provisioning an AKS cluster also provisions a load balancer, so we'll want a subresource for [Azure Load Balancers](https://azure.microsoft.com/en-us/pricing/details/load-balancer/) The `network_profile.load_balancer_sku` tells us if it's standard or basic.
* If `http_application_routing.enabled == true` we want to add a subresource [Azure DNS pricing](https://azure.microsoft.com/en-gb/pricing/details/dns/#pricing)
https://azure.microsoft.com/en-us/pricing/details/load-balancer/
* If `oms_agent.enabled == true` we'll also want a subresource for [Azure Monitoring pricing](https://azure.microsoft.com/en-gb/pricing/details/monitor/)

**Comment by ekifel** [comment]
@alikhajeh1, hey, I want to work on it!

**Comment by alikhajeh1** [comment]
@ekifel the master branch test log output is showing this, can you please check?
```
time="2021-05-26T17:15:48Z" level=warning msg="No products found for default_node_pool Instance usage (pay as you go, D2S v2), using 0.00"

```

**Comment by alikhajeh1** [comment]
Shipped in v0.8.7 - @NissesSenap let us know if you find any issues with it!

**Comment by alikhajeh1** [comment]
The "future" is being handled as part of https://github.com/infracost/infracost/issues/710
